iroh/src/config.rs

```diff
@@ -22,6 +23,55 @@ pub const CONFIG_FILE_NAME: &str = "iroh.config.toml";
 /// For example, `IROH_PATH=/path/to/config` would set the value of the `Config.path` field
 pub const ENV_PREFIX: &str = "IROH";
 
+/// Paths to files or directory within the [`iroh_data_root`] used by Iroh.
+pub enum IrohPaths {
+    /// Path to the node's private key for the [`crate::net::PeerId`]
+    Keypair,
+    /// Path to the node's flat-file bao blob store ([`crate::baomap::flat`])
+    BaoFlatStore,
+}
+impl From<&IrohPaths> for &'static str {
+    fn from(value: &IrohPaths) -> Self {
+        match value {
+            IrohPaths::Keypair => "keypair",
+            IrohPaths::BaoFlatStore => "blobs.v0",
+        }
+    }
+}
+impl FromStr for IrohPaths {
+    type Err = anyhow::Error;
+    fn from_str(s: &str) -> Result<Self> {
+        Ok(match s {
+            "keypair" => Self::Keypair,
+            "blobs.v0" => Self::BaoFlatStore,
+            _ => bail!("unknown file or directory"),
+        })
+    }
+}
+impl fmt::Display for IrohPaths {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let s: &str = self.into();
+        write!(f, "{s}")
+    }
+}
+impl AsRef<Path> for IrohPaths {
+    fn as_ref(&self) -> &Path {
+        let s: &str = self.into();
+        Path::new(s)
+    }
+}
+impl IrohPaths {
+    /// Get the path for this [`IrohPath`] by joining the name to `IROH_DATA_DIR` environment variable.
+    pub fn env_path(self) -> Result<PathBuf> {
+        let mut root = iroh_data_root()?;
+        if !root.is_absolute() {
+            root = std::env::current_dir()?.join(root);
```

Comment:
shouldn't this use `https://doc.rust-lang.org/std/path/struct.Path.html#method.canonicalize` instead?

Reply:
This comes from current main, I only copied it from `src/commands/provide.rs`: https://github.com/n0-computer/iroh/pull/1345/files#diff-7de268e3454ececdfb93202f8f3bb012d38673ddcdca12fd5f7c5e2a6778dc5bL53 
so dunno for sure. Does `canonicalize` join to `env::current_dir()` for relative paths? And - is that actually what we want?

Reply:
hmm, okay lets leave it as is for now